Various lint fixes

pjonsson opened the pull request:
Enable the bugbear rules, and most of the Ruff rules.

<!-- readthedocs-preview datacube-ows start -->
----
📚 Documentation preview 📚: https://datacube-ows--1156.org.readthedocs.build/en/1156/

<!-- readthedocs-preview datacube-ows end -->

pjonsson committed:
Remove unused loop variables

pjonsson committed:
tests: stop asserting False

Use the expected exception mechanism
in the tests instead of try-except
and then asserting False in
the wrong branch.

This fixes some bugbear warnings,
but it should also give better
error messages from pytest
if the wrong exception is thrown.

pjonsson committed:
Remove mutable default arguments

The argument is shared across
all calls to the function, so
modifications persist across
calls.

pjonsson committed:
Re-raise exceptions from something

These all look like they should be
from None, but that is a coincidence.

pjonsson committed:
ruff: enable bugbear rules

pjonsson committed:
Remove unused noqa directives

pjonsson committed:
ruff: enable RUF rules

Enable most of the Ruff rules.

SpacemanPaul commented on datacube_ows/config_utils.py:
```diff
@@ -43,7 +43,7 @@
 # If modification of inclusions is a required, a copy (ninclusions) is made and modified instead.
 # pylint: disable=dangerous-default-value
 def cfg_expand(cfg_unexpanded: CFG_DICT,
-               cwd: str | None = None, inclusions: list[str] = []) -> CFG_DICT:
```
If we're going to change this to None, we should remove the comment immediately above that explains why `[]` is actually safe in this case.

pjonsson replied:
Wow, strong case of tunnel vision on my behalf, I've fixed it in #1157.

pjonsson committed:
Remove dead code

I thought these were with lazy
configuration loading, but it turns
out to have been leftovers, so remove
them.

pjonsson committed:
tests: get first element lazily

Do not force the entire computation
to then select the first element
of the list, just get the next
element from the iterator instead.

pjonsson committed:
Use f-string conversion operators

Use "e!s" instead of "str(e)" inside
f-strings.

pjonsson committed:
tests: call Decimal() with string

This avoids rounding errors.

pjonsson committed:
Remove unnecessary int conversions

These already return integers.

pjonsson committed:
wcs1_utils: avoid key check

pjonsson committed:
ogc_exceptions: use unpacking operator

Unpacking is more efficient than concatenation
and also works on iterables.
